### ISSUE TITLE
[MCOMPILER-470] -parameters doesn't work with --release

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-compiler-plugin#691]
**[Daniel Shuy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=daniel-shuy)** opened **[MCOMPILER-470](https://issues.apache.org/jira/browse/MCOMPILER-470?redirect=false)** and commented

The **parameters** flag works when configuring the Java version with **source** and **target**, eg.

```java
<configuration>
  <source>11</source>
  <target>11</target>
  <parameters>true</parameters>
</configuration>
```

However, when configuring the Java version with **release**, as is the convention to use with Java 9+, the **parameters** flag no longer works (the **.class** file doesn't contain **MethodParameters**), eg.

```java
<configuration>
  <release>11</release>
  <parameters>true</parameters>
</configuration>
```

 

To ensure that the issue is not with **javac**, I've tested running **javac** with:

```java
javac --source 11 --target 11 -parameters MyClass.java
```

and

```java
javac --release 11 -parameters MyClass.java
```

and verified that it works for both (both **.class** files contain **MethodParameters**).

---

**Affects:** 3.8.1

**Backported to:** [3.9.0](https://github.com/apache/maven-compiler-plugin/milestone/29?closed=1)

**Comment by jira-importer** [comment]
**[Olivier Lamy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=olamy)** commented

Fixed with plexus compiler upgrade

**Comment by jira-importer** [comment]
**[Falko Modler](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=famod)** commented

[Olivier Lamy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=olamy) are you sure this isn't already fixed with 3.9.0? I was able to remove my wordarounds for this when I updated to 3.9.0 a few weeks ago...
